### ISSUE TITLE
graphics_card_resource.py couldn't define index for iGPU and dGPU correctly

### ISSUE THREAD

**Issue by hanhsuan** [canonical/checkbox#491]
### Bug Description

graphics_card_resource.py will use [PCI ID](https://github.com/canonical/checkbox/blob/main/providers/resource/bin/graphics_card_resource.py#L95) to sort the GPU order. However we get one new [DUT](https://certification.canonical.com/hardware/202304-31460/) with `AMD(APU)` + `NV(dGPU)`, the index will be set to `2` for `AMD(APU)` and `1` for `NV(dGPU)`. 
* Advanced Micro Devices, Inc. [AMD/ATI] - 1002:15bf
* nVidia - 10de:28a1

Finally create the wrong test case
* [graphics/1_glxgears_PCI_ID_0x28a1](https://certification.canonical.com/hardware/202304-31460/submission/313703/test/198597/result/33368789/)
* [graphics/2_glxgears_PCI_ID_0x15bf](https://certification.canonical.com/hardware/202304-31460/submission/313703/test/198698/result/33368797/)

and doing the wrong prime-offload to AMD(APU).
```
Setting up PRIME GPU offloading for AMD discrete GPU
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
2038 frames in 5.0 seconds = 406.658 FPS
1387 frames in 5.0 seconds = 276.908 FPS
```

In the future, there might be more combination of GPU, such as:
* Intel(iGPU) +  Intel(dGPU)
* AMD(APU) +  NV(dGPU)
* AMD(APU) + Intel(dGPU)

Therefore, we need a way to define the index correctly.

### To Reproduce

Run graphic test

### Environment

- OS: Ubuntu 22.04.2 LTS
- Checkbox type: Debian
- Checkbox version: 2.5
- CPU: AMD APU
- GPU: Nvidia

**Comment by pieqq** [comment]
Before addressing this bug, I think we should have a discussion to decide how we should test grpahics card with more recent versions of Ubuntu. The current system used by Checkbox was developed at different time, back when

- nvidia GPU did not have on-demande mode
- there was no AMD APUs being certified
- there was no Intel dGPUs on the market
- etc.


I'll prepare a Jira card to track this.

**Comment by pieqq** [comment]
Fixed with #586 and #942
